hazelcast/src/main/java/com/hazelcast/cluster/Address.java

```diff
@@ -185,4 +186,22 @@ private static InetAddress resolve(InetSocketAddress inetSocketAddress) {
         }
         return address;
     }
+
+
+    /**
+     * Creates an unresolved address. This API is used by the client for the case where the hostname can not be resolved.
+     * Members achieve the same thing via Serialization where we don't resolve the hostname when Address comes as Data
+     * from another member.
+     *
+     * @param host hostname or IP
+     * @param port port
+     * @return an unresolved address
+     */
+    @PrivateApi
+    public static Address createUnresolvedAddress(String host, int port) {
```

Comment:
So IIC when an address is deserialized and created by the `CustomTypeFactory` we store the raw `host` value without trying to resolve it. The PR says "postpone"... so where is it finaly resolved?

Reply:
We rely on the fact that it is already being re-resolved again on connect. See 
https://github.com/hazelcast/hazelcast/blob/35b9b7c41a9fe50c8ceb76915146181a526e1906/hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java#L709
which makes a call to Address.getInetAdress here:
https://github.com/hazelcast/hazelcast/blob/35b9b7c41a9fe50c8ceb76915146181a526e1906/hazelcast/src/main/java/com/hazelcast/cluster/Address.java#L101